speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -39,162 +40,445 @@
+    """
+    config.output_hidden_states = True  # We want the hidden states as well!
+    return config
+
+
+def model_set_spectral_augmentation(model, apply_spec_augment):
+    """Sets `model.config.apply_spec_augment` the flag to a specific value.
+    """
+    model.config.apply_spec_augment = apply_spec_augment
+    return model
+
+
+def modify_state_dict_wav2vec2(path):
+    """Function to be used in HuggingFaceModel._load_sb_pretrained_parameters.
+
+    A custom loading ensures SpeechBrain compatibility for Pretrain and model
+    de/serialization. Here, '.wav2vec2' is removed before loading.
+    """
+    modified_state_dict = {}
+    orig_state_dict = torch.load(path, map_location="cpu")
+
+    # We remove the .wav2vec2 in the state dict.
+    for key, params in orig_state_dict.items():
+        if "wav2vec2." in key:
+            save_key = key.replace("model.wav2vec2.", "")
+            modified_state_dict[save_key] = params
+
+    return modified_state_dict
+
+
+def default_forward(model, data):
+    """Takes input data and returns its forward pass of a given model.
+     To be used in: HuggingFaceModel._forward.
+
+    Arguments
+    ---------
+    model : transformers.AutoModel
+        A valid HuggingFace transformers model.
+    data : torch.Tensor (signal)
+        A batch of audio signals to transform to features.
+
+    Returns
+    -------
+    out : torch.Tensor
+        Batch of depending model outputs
+    norm_shape : List[int]
+        Shape to be used in layer norm.
+    """
+    out = model(data)
+    norm_shape = out.shape
+    return out, norm_shape
+
+
+def wav2vec2_forward(model, data, output_all_hiddens):
+    """Takes an input waveform and return its corresponding wav2vec encoding.
+
+    Arguments
+    ---------
+    model : transformers.AutoModel
+        A valid HuggingFace transformers model.
+    data : torch.Tensor (signal)
+        A batch of audio signals to transform to features.
+
+    Returns
+    -------
+    out : torch.Tensor
+        Batch of depending model outputs
+    norm_shape : List[int]
+        Shape to be used in layer norm.
+    """
+    # Extract wav2vec output
+    out = model(data, output_hidden_states=True)
+
+    if output_all_hiddens:
+        out = torch.stack(list(out.hidden_states), dim=0)
+        norm_shape = out.shape[-3:]
+    else:
+        out = out.last_hidden_state
+        norm_shape = out.shape
+
+    return out, norm_shape
+
+
+def wav2vec2_pretraining_forward(model, data, mask_prob, mask_length):
+    """Takes an input waveform and return its corresponding wav2vec encoding.
+
+    Parameters
+    ----------
+    model : transformers.AutoModel
+        A valid HuggingFace transformers model.
+    data : torch.Tensor (signal)
+        A batch of audio signals to transform to features.
+    mask_prob : float (default: 0.65)
+        Probability of masking a given frame. Default is taken from the paper.
+    mask_length : float (default: 10)
```

Comment:
no default value in the function

Reply:
it's there - please see ;-)
```
        mask_prob=0.65,
        mask_length=10,
```

Reply:
got what you mean; just copy pasted to docstring - well, then I'd remove the default here

Reply:
handling a default value there is too much; this function will get called way too often for that